src/script/sign.cpp

```diff
@@ -86,16 +92,21 @@ static bool SignStep(const SigningProvider& provider, const BaseSignatureCreator
     case TX_NULL_DATA:
     case TX_WITNESS_UNKNOWN:
         return false;
-    case TX_PUBKEY:
-        if (!creator.CreateSig(provider, sig, CPubKey(vSolutions[0]).GetID(), scriptPubKey, sigversion)) return false;
+    case TX_PUBKEY: {
+        CPubKey pubkey(vSolutions[0]);
+        keyID = pubkey.GetID();
+        if (!creator.CreateSig(provider, sig, keyID, scriptPubKey, sigversion)) return false;
+        sigdata.signatures.emplace(keyID, SigPair(pubkey, sig));
```

Comment:
Why is this needed? Signing a TX_PUBKEY should always result in a full signature.

Reply:
A PSBT does not have to create the final scriptSig even when a complete set of signatures is available. The user has the option to not finalize the scriptSigs even when they can be.

Reply:
In that case you should also be able to deal with unfinalized signatures already present, and turn them into scriptSigs. In other words, you need to test whether sigdata.signatures contains a matching signature and reuse it. If that's the case, we're probably better off reintroducing SignatureDataSignatureCreator and SignatureDataSigningProvider, so these keys/script/signatures are automatically available everywhere.

Another interpretation is that TX_PUBKEY and TX_PUBKEYHASH etc just don't have "partial" forms - they're either fully signed or not, and as a result there never is anything to finalize.

Reply:
We should discuss this on the mailing list as it requires changing the BIP.

For now, I have added a helper function for createSig which fetches and places signatures to and from the SignatureData.